crates/ruff_linter/src/rules/flake8_builtins/rules/builtin_module_shadowing.rs

```diff
@@ -42,10 +43,10 @@ pub(crate) fn builtin_module_shadowing(
     allowed_modules: &[String],
     target_version: PythonVersion,
 ) -> Option<Diagnostic> {
-    if !path
-        .extension()
-        .is_some_and(|ext| ext == "py" || ext == "pyi")
-    {
+    if !matches!(
+        PySourceType::try_from_path(path),
+        Some(PySourceType::Python | PySourceType::Stub)
```

Comment:
Nit: This seems somewhat common. We could consider introducing a method for it: `is_python_or_stub`

Reply:
makes sense